crates/snapshot/src/snapshotter.rs

```diff
@@ -79,21 +83,37 @@ impl SnapshotTargets {
 
 impl<DB: Database> Snapshotter<DB> {
     /// Creates a new [Snapshotter].
-    pub fn new(db: DB, chain_spec: Arc<ChainSpec>, block_interval: u64) -> Self {
+    pub fn new(
+        db: DB,
+        snapshots_path: PathBuf,
+        chain_spec: Arc<ChainSpec>,
+        block_interval: u64,
+    ) -> RethResult<Self> {
+        // Create directory for snapshots if it doesn't exist.
+        if !snapshots_path.exists() {
+            std::fs::create_dir_all(&snapshots_path).map_err(|e| {
```

Comment:
I think `reth_primitives::fs::create_dir_all` can be used here for a nice error

Reply:
had to add a From impl for `RethError` tho, but yea